core-js/src/main/javascript/cdf/components/ccc/BaseCccComponent.js

```diff
@@ -14,52 +14,147 @@
 define([
   '../ChartComponent',
   '../../lib/CCC/pvc',
+  '../../lib/CCC/def',
   '../../lib/modernizr',
   '../../lib/jquery',
-  '../../lib/CCC/protovis-compat!'
-], function(ChartComponent, pvc, Modernizr, $, pv) {
+  'amd!../../lib/underscore',
+  '../../lib/CCC/protovis-compat!',
+  '../../PentahoTypeContext',
+  'pentaho/visual/color/paletteRegistry',
+  'pentaho/shim/es6-promise'
+], function(ChartComponent, pvc, def, Modernizr, $, _, pv, PentahoTypeContext, paletteRegistry, Promise) {
 
-  pvc.defaultCompatVersion(2);
+  pvc.defaultCompatVersion(3);
+
+  var _reCccName = /(.*)(Chart)$/g;
+
+  var _context = PentahoTypeContext.getInstance();
+
+  var _chartTypesBlackList = ['boxplot', 'bullet', 'treemap', 'waterfall'];
+  var _chartFormExceptions = ['heatGrid', 'line', 'areaStacked', 'scatter', 'pie', 'pointAbstract', 'sunburst'];
+  var _valuesNormalizedExceptions = ['barNormalized'];
+  var _stackedExceptions = ['barNormalized'];
 
   return ChartComponent.extend({
 
     query: null,
     chart: null,
 
-    _preProcessChartDefinition: function() {
+    _cccChartName: null,
+    getCccChartName: function () {
+      if (!this._cccChartName && this.cccType) {
+        this._cccChartName = this._getCccFullName(def.qualNameOf(this.cccType).name);
+      }
+      return this._cccChartName;
+    },
+
+    /**
+     *
+     * @param name
+     * @returns {*}
+     * @private
+     */
+    _getCccFullName: function (name) {
+      var fullName;
+      //get the ccc name first
+      var m = name.split(_reCccName);
```

Comment:
Use of split to do this doesn't seem right — not the right tool for the job, even if you can get it to do what you want. For this use case, split returns an array of 4 entries...